lua/dap/utils.lua

```diff
@@ -0,0 +1,68 @@
+local M = {}
+
+
+local function iscallable(x)
+  if type(x) == 'function' then
+    return true
+  elseif type(x) == 'table' then
+    local mt = debug.getmetatable(x)
+    return type(mt) == "table" and type(mt.__call) == "function"
+  else
+    return false
+  end
+end
+
+
+function M.map_to_key(fn, tbl)
+  local rtn = {}
+  local callable = iscallable(fn)
```

Comment:
Looks like currently all call-sites always pass in a function. I think the `iscallable` could be removed for now and we can have the contract that `fn` needs to be a function.